maud_extras/lib.rs

```diff
@@ -30,3 +30,31 @@ impl<T: AsRef<str>> Render for Css<T> {
         }
     }
 }
+
+/// Links to an external javascript.
+///
+/// # Example
+///
+/// ```rust
+/// # #![feature(plugin)]
+/// # #![plugin(maud_macros)]
+/// # extern crate maud;
+/// # extern crate maud_extras;
+/// # use maud_extras::*;
+/// # fn main() {
+/// let markup = html! { (Javascript("app.js")) };
+/// assert_eq!(markup.into_string(),
+///            r#"<script src="app.js" type="text/javascript"></script>"#);
+/// # }
+/// ```
+pub struct Javascript<T: AsRef<str>>(pub T);
+
+impl<T: AsRef<str>> Render for Javascript<T> {
+    fn render(&self) -> Markup {
+        html! {
+            script src=(self.0.as_ref()) type=("text/javascript") {}
```

Comment:
This might not work if the string has special characters as those need to be percent encoded rather than escaped in a HTML fashion. 
https://url.spec.whatwg.org/#url-code-points